Implement BlockChain<T>.FindBranchPoint again,  to reduce memory consumption (#299)
planetarium/libplanet: CHANGES.md

 -  Added IPv6 support to `Libplanet.Stun.StunAddress`. [[#267], [#271]]
 -  `IStore.GetBlockStates()` became able to return `null` to represent an absence
    of states (i.e., incomplete states).  [[#272], [#285]]

>  -  `BlockChain<T>.FindBranchPoint` became to use memory less than before. [[#282], [#299]]

Reviewer: ```suggestion
 -  `BlockChain<T>.FindBranchPoint()` became to use less memory than before.
    [[#282], [#299]]
```

Author: I changed it :ok_hand:

    an empty `TxIds` from peers, and it had made the network waste bandwidth for
    unnecessary messages.  `Swam` became to no more send such empty `GetTxs`.
    [[#297]]

>  -  `BlockChain<T>.FindBranchPoint()` became to use less memory than before.

Reviewer: ```suggestion
 -  Reduced the memory footprint of `BlockChain<T>.FindBranchPoint()` method
    under the circumstances that the height of the `BlockChain<T>` object
```

>     [[#282], [#299]]

Reviewer: ```suggestion
    is high.  [[#282], [#299]]
```

    an empty `TxIds` from peers, and it had made the network waste bandwidth for
    unnecessary messages.  `Swam` became to no more send such empty `GetTxs`.
    [[#297]]
 -  Reduced the memory footprint of `BlockChain<T>.FindBranchPoint()` method

>     under the circumstances that the height of the `BlockChain<T>` object is high.  [[#282], [#299]]

Reviewer: Although it's quite trivial, this single line seems too long.  Could you split this line into two?  The maximum width is 80 columns.
